### DISCUSSION TITLE
Horizon Failed?.

### DISCUSSION THREAD

**Discussion by jhemmmm** [spatie/laravel-rate-limited-job-middleware#31] [Q&A]
My job keeps failing.
```
Illuminate\Queue\MaxAttemptsExceededException: App\Notifications\Replied has been attempted too many times or run too long. The job may have previously timed out. in vendor/laravel/framework/src/Illuminate/Queue/Worker.php:648
```

```
   /**
     * Middleware RateLimiter
     */
    public function middleware()
    {
        $rateLimitedMiddleware = (new RateLimited())
            ->allow(1)
            ->everySeconds(10);

        return [$rateLimitedMiddleware];
    }
}
```

```
 'supervisor-4' => [
                'connection' => 'redis',
                'queue' => ['notification'],
                'balance' => 'simple',
                'processes' => 1,
                'timeout' => 600,
                'tries' => 1,
                'nice' => 0,
            ],
            ```

**Comment by mashirozx** [comment]
same issue here

**Comment by chicouskiluiz** [comment]
same thing here, any solution?

**Comment by mashirozx** [comment]
TL;DR
It's not recommended to use muddleware here, try to use redis throttle directly (u can understand this middleware as a Redis:: throttle wrapper actually).

***
As @jhemmmm 's code, you just need to increase the `tries` of queue and the `timeout` of horizon.

But you can do it more easily, like this: 
https://github.com/xflash-panda/v2board/commit/69e9b730ad69bca986fac89e52597b542ecade78#diff-74d05b3f3dc9b1bb6eccdf41b2ecc2d2e72ca671e6cc29606a2d51fe5a4265a8R21

**Reply by jhemmmm** [reply]
adding tries and timeout didn't work, tried it before, and the job will be automatically failed in an instant. I have fixed this by using another library.

**Reply by FelipeVeiga** [reply]
how did you fixed ?

**Comment by FelipeVeiga** [comment]
same here.
